lib/logstash/filters/geoip.rb

```diff
@@ -153,8 +152,15 @@ def setup_filter(database_path)
   end
 
   # call by DatabaseManager
-  def expire_action
-    fail_filter
+  def update_filter(action, *args)
+    case
+    when action == :update
+      setup_filter(*args)
+    when action == :expire
+      fail_filter
+    else
+      @logger.warn("invalid callback", :action => action)
+    end
   end
```

Comment:
minor ruby style suggestion:

```suggestion
    case action
    when :update
      setup_filter(*args)
    when :expire
      fail_filter
    else
      @logger.warn("invalid callback", :action => action)
    end
  end
```